<launch>
<!-- @Hover thrust annotation -->
    <!-- Typhoon480 SITL is 0.4455 -->
    <!-- Aims-Hexa-v3 is 0.40 -->
    <!-- Iris SITL is 0.70 -->
    <!-- Aims-Quad-Gamma is 0.365 -->
  
    <node pkg="aims_als" name="pos_ctrl_node" type="pos_ctrl_exe" output="screen">
        <param name="Hz_of_MONITOR" value="1.0"/>

        <param name="feedforward" value="true" />
        <param name="DOB"         value="false" />
        <param name="debug"       value="false" />
        <param name="sitl"        value="false" />
        <param name="logging"     value="true" />
        <param name="fixed_yaw"   value="false" />

        <param name="uav_vel_lpf_freq"  value="1.5" />
        <param name="uav_acc_lpf_freq"  value="5.0" />
        <param name="uav_hover_thrust"  value="0.43" />

        <param name="gv_vel_lpf_freq"  value="1.5" />
        <param name="gv_head_lpf_freq" value="1.0" />
        <param name="gv_fixed_head"    value="90.0" />

        <param name="standoff_x"  value="-0.4" />
        <param name="standoff_y"  value="0.0" />
        <param name="standoff_z"  value="1.0" />
        <param name="a_cmd_alpha" value="1.0" />

        <param name="q_filter_lpf_freq" value="0.1" />
        <param name="time_constant_of_uav" value="0.7" />

        <!-- <param name="vf_hyper"   value="false" />    
        <param name="vf_phi_des"   value="-27.5" />
        <param name="vf_phi_delta" value="34.375" />
        <param name="vf_r_max"     value="5.0" />
        <param name="vf_k1"        value="0.25" />
        <param name="vf_k2"        value="0.25" />
        <param name="vf_margin"    value="17.1888" />
        <param name="vf_c1"        value="8.5" />
        <param name="vf_c2"        value="2.0" />
        <param name="vf_n1"        value="4.0" />
        <param name="vf_n2"        value="3.0" />
        <param name="vf_gamma"     value="2.5" />
        <param name="vf_moving"    value="false" /> -->

        <param name="vf_hyper"     value="true" />    
        <param name="vf_phi_des"   value="-27.5" />
        <param name="vf_phi_delta" value="34.375" />
        <param name="vf_r_max"     value="5.0" />
        <param name="vf_k1"        value="0.25" />
        <param name="vf_k2"        value="0.20" />
        <param name="vf_margin"    value="17.1888" />
        <param name="vf_c1"        value="8.5" />
        <param name="vf_c2"        value="2.0" />
        <param name="vf_n1"        value="2.0" />
        <param name="vf_n2"        value="2.0" />
        <param name="vf_gamma"     value="2.5" />
        <param name="vf_moving"    value="false" />


        <param name="saturation_thrust_min"          value="0.35" />
        <param name="saturation_thrust_max"          value="0.65" />
        <param name="saturation_tilt"                value="40.0" />
        <param name="saturation_const_dt"            value="0.02" />
        <param name="saturation_vel_xy_max"          value="3.0" />
        <param name="saturation_vel_z_max"           value="1.0" />
        <param name="saturation_integral_vel_xy_max" value="5.0" />
        <param name="saturation_integral_vel_z_max"  value="3.0" />
        <param name="saturation_no_meas_duration"    value="0.1" />
        <param name="saturation_critical_altitude"   value="0.33" />
        <param name="saturation_land_point_altitude" value="-0.0" />
0

        <param name="vx_kp" value="2.2" />
        <param name="vx_ki" value="0.8" />
        <param name="vx_kd" value="0.0" />

        <param name="vy_kp" value="2.2" />
        <param name="vy_ki" value="0.8" />
        <param name="vy_kd" value="0.0" />

        <param name="vz_kp" value="4.0" />
        <param name="vz_ki" value="2.0" />
        <param name="vz_kd" value="0.0" />

        <param name="px_kp" value="0.90" />
        <param name="px_kd" value="0.10" />
        
        <param name="py_kp" value="1.10" />
        <param name="py_kd" value="0.2" />
        
        <param name="pz_kp" value="1.2" />
        <param name="pz_kd" value="0.10" />

    </node>

     <node pkg="aims_als" name="led_ctrl_node" type="led_ctrl.py" output="screen" />

</launch>
